lib/spack/spack/main.py

```diff
@@ -925,6 +925,9 @@ def _main(argv=None):
             env = spack.cmd.find_environment(args)
             if env:
                 ev.activate(env, args.use_env_repo)
+                # move command line scope to the highest level again
+                command_line_scope = spack.config.CONFIG.scopes.pop("command_line")
+                spack.config.CONFIG.push_scope(command_line_scope)
```

Comment:
This diff doesn't seem to fail locally:
```diff
diff --git a/lib/spack/spack/config.py b/lib/spack/spack/config.py
index afd8f30bac..66713f90af 100644
--- a/lib/spack/spack/config.py
+++ b/lib/spack/spack/config.py
@@ -752,10 +752,6 @@ def override(
         assert scope is overrides
 
 
-#: configuration scopes added on the command line set by ``spack.main.main()``
-COMMAND_LINE_SCOPES: List[str] = []
-
-
 def _add_platform_scope(cfg: Configuration, name: str, path: str, writable: bool = True) -> None:
     """Add a platform-specific subdirectory for the current platform."""
     platform = spack.platforms.host().name
@@ -860,13 +856,6 @@ def create() -> Configuration:
         # Each scope can have per-platfom overrides in subdirectories
         _add_platform_scope(cfg, name, path)
 
-    # add command-line scopes
-    _add_command_line_scopes(cfg, COMMAND_LINE_SCOPES)
-
-    # we make a special scope for spack commands so that they can
-    # override configuration options.
-    cfg.push_scope(InternalConfigScope("command_line"))
-
     return cfg
 
 
diff --git a/lib/spack/spack/main.py b/lib/spack/spack/main.py
index cd002f778a..7cab47d77f 100644
--- a/lib/spack/spack/main.py
+++ b/lib/spack/spack/main.py
@@ -911,13 +911,6 @@ def _main(argv=None):
     # Make spack load / env activate work on macOS
     restore_macos_dyld_vars()
 
-    # make spack.config aware of any command line configuration scopes
-    if args.config_scopes:
-        spack.config.COMMAND_LINE_SCOPES = args.config_scopes
-
-    # ensure options on spack command come before everything
-    setup_main_options(args)
-
     # activate an environment if one was specified on the command line
     env_format_error = None
     if not args.no_env:
@@ -925,15 +918,18 @@ def _main(argv=None):
             env = spack.cmd.find_environment(args)
             if env:
                 ev.activate(env, args.use_env_repo)
-                # move command line scope to the highest level again
-                command_line_scope = spack.config.CONFIG.scopes.pop("command_line")
-                spack.config.CONFIG.push_scope(command_line_scope)
         except spack.config.ConfigFormatError as e:
             # print the context but delay this exception so that commands like
             # `spack config edit` can still work with a bad environment.
             e.print_context()
             env_format_error = e
 
+    # Push scopes from the command line last
+    if args.config_scopes:
+        spack.config._add_command_line_scopes(spack.config.CONFIG, args.config_scopes)
+    spack.config.CONFIG.push_scope(spack.config.InternalConfigScope("command_line"))
+    setup_main_options(args)
+
     # ------------------------------------------------------------------------
     # Things that require configuration should go below here
     # ------------------------------------------------------------------------

```
and would make the behavior uniform also with `-C`. Let me know if that is preferable and I'll push it.

Reply:
Pushed